src/nvim/ex_getln.c

```diff
@@ -214,6 +215,9 @@ static int hislen = 0;                  /* actual length of history tables */
 /// user interrupting highlight function to not interrupt command-line.
 static bool getln_interrupted_highlight = false;
 
+/// Pointer to flag which keeps track if mouse was used
+static int *mouse_used_p = NULL;
```

Comment:
I think we should avoid storing pointers if not necessary. Instead set the return value in `getcmdline_prompt` after `getcmdline`. Also place the value inside `ccline`, not globally. (if mouse support is conditional, an explicit bool variable is also better than a NULL check far from where the pointer is defined)